PyInstaller/utils/hooks/qt.py

```diff
@@ -175,19 +199,19 @@ def qt_menu_nib_dir(namespace):
     """
     Return path to Qt resource dir qt_menu.nib on OSX only.
 
-    :param namespace: Import namespace, i.e., PyQt5 or PySide2
+    :param namespace: Import namespace, i.e., PyQt5 or PySide2/6
 
     :return: Directory containing qt_menu.nib for specified namespace
     """
-    if namespace not in ['PyQt5', 'PySide2']:
+    if namespace not in Qt_Librairies:
         raise Exception('Invalid namespace: {0}'.format(namespace))
     menu_dir = None
 
     path = exec_statement("""
     from {0}.QtCore import QLibraryInfo
-    path = QLibraryInfo.location(QLibraryInfo.LibrariesPath)
+    path = QLibraryInfo.location(QLibraryInfo.LibrariesPath{1})
     print(path)
-    """.format(namespace))
+    """.format(namespace, '()' if namespace == 'PySide6' else ''))
```

Comment:
Why is this change necessary? As far as I can tell `QLibraryInfo.LibrariesPath` is not callable on `PySide6` either, and calling `qt_menu_nib_dir('PySide6')` raises an exception about it:
```python
>>> from PyInstaller.utils.hooks.qt import qt_menu_nib_dir
>>> qt_menu_nib_dir('PySide6')
Traceback (most recent call last):
  File "<string>", line 3, in <module>
TypeError: 'PySide6.QtCore.QLibraryInfo.LibraryPath' object is not callable
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/[...]/venv/lib/python3.7/site-packages/PyInstaller/utils/hooks/qt.py", line 219, in qt_menu_nib_dir
    """.format(namespace, ", ".join(paths)))
Exception: 
            Cannot find qt_menu.nib for PySide6
            Path checked: Resources, QtGui.framework/Resources, /Users/rok/Development/pyi-pyside6/venv/python.app/Contents/Resources
```

Reply:
I totally agree with the fact that it's a bit strange as all other `QLibraryInfo` attributes aren't callable
However, using the latest version of PySide6 available with `pip3 install --upgrade PySide6` on macOS / Ubuntu 20.04LTS gives

```python3
>>> from PySide6 import QtCore
>>> QtCore.QLibraryInfo.LibraryPath
<class 'PySide6.QtCore.QLibraryInfo.LibraryPath'>
>>> QtCore.QLibraryInfo.LibraryPath()
PySide6.QtCore.QLibraryInfo.LibraryPath.PrefixPath
```

Otherwise I get
```python3
>>> QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.LibraryPath)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'PySide6.QtCore.QLibraryInfo.location' called with wrong argument types:
  PySide6.QtCore.QLibraryInfo.location(EnumType)
Supported signatures:
  PySide6.QtCore.QLibraryInfo.location(PySide6.QtCore.QLibraryInfo.LibraryPath)
```

Python versions
```
Python 3.9.1 (default, Dec 17 2020, 10:08:12)
[Clang 12.0.0 (clang-1200.0.32.27)] on darwin
```
 and 
```
Python 3.8.5 (default, Jul 28 2020, 12:59:40)
[GCC 9.3.0] on linux
```

Pyside6 is reporting version 6.0.0

**Maybe it's a PySide6 related issue !**
Could you try to upgrade with pip and test on your platforms ?

Reply:
The function in question is using 
```python
QLibraryInfo.location(QLibraryInfo.LibrariesPath)
```
and not
```python
QLibraryInfo.location(QLibraryInfo.LibraryPath)
```
One is callable, the other not, and they resolve to different locations:

```python
>>> from PySide6 import QtCore
>>> QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.LibrariesPath)
'/Users/[...]/venv/lib/python3.7/site-packages/PySide6/Qt/lib'
>>> QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.LibraryPath())
'/Users/[...]/venv/lib/python3.7/site-packages/PySide6/Qt'
```

Neither PySide2 nor PySide6 appear to contain the `qt_menu.nib` file that this function searches for, but based on the search paths (`os.path.join(path, 'QtGui.framework', 'Resources')`) I suspect that `QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.LibrariesPath)` is correct.

Reply:
Although FWIW, `qt_menu_nib_dir` does not seem to be used anywhere, anyway, so we might consider removing it...

Reply:
LGTM, fixed in https://github.com/pyinstaller/pyinstaller/pull/5418/commits/3d16edde27e11947d18ebf09d9ba5e41e2a36768 . I have not removed `qt_menu_nib_dir` function as I don't know the implications and this might be out of this PR's scope